### PR TITLE
Save zone state

### PR DESCRIPTION
**Pull request by slyjeff**
Saving a file will now save the state of which cards are in which zones.

### CODE REVIEW

File: ArkhamOverlay/Data/Game.cs [hunk 1]
```diff
@@ -141,6 +141,24 @@ public class Game : ViewModel, IGame {
...
+                _logger.LogError($"Cannot add card {button.CardInfo.Name} to {destinationCardGroup.Name} because zone with index {zoneIndex} does not exist");
+                return;
+            }
+
+            _logger.LogMessage($"Adding card {button.CardInfo.Name} to {destinationCardZone.Name} of {destinationCardGroup.Name} ");
+            destinationCardZone.CreateCardButton(button, CreateButtonOptions(destinationCardGroup, destinationCardZone, button.CardInfo));
+        }
+
         /// <summary>
         /// When a button is clicked, look at the context, see if it's a button in this card group, and execute the appropraite logic
```
Comment (davilimap) [inline ArkhamOverlay/Data/Game.cs]: Nit: Not part of this PR, but *appropriate

File: ArkhamOverlay/Services/GameFileService.cs [hunk 2]
```diff
@@ -16,23 +19,41 @@ public interface IGame {
...
+    /// Information about a button in a zone for saving and recreating state
+    /// </summary>
+    public class ZoneButton {
+        /// <summary>
+        /// Card Group the button blongs to
+        /// </summary>
+        public CardGroupId CardGroupId { get; set; }
+
+        /// <summary>
+        /// Zone in the card group that the button blongs to
```
Comment (davilimap) [inline ArkhamOverlay/Services/GameFileService.cs]: Minor: belongs

File: ArkhamOverlay/Services/GameFileService.cs [hunk 3]
```diff
@@ -16,23 +19,41 @@ public interface IGame {
...
+        /// </summary>
+        public int ZoneIndex { get; set; }
+
+        /// <summary>
+        /// Identify the button by name
```
Comment (davilimap) [inline ArkhamOverlay/Services/GameFileService.cs]: Super nit: this description is an action, change to a subject (e.g. Name of the button)

File: ArkhamOverlay/Services/GameFileService.cs [hunk 4]
```diff
@@ -95,11 +119,15 @@ public class GameFileService {
         internal void Save(string fileName) {
             _logger.LogMessage($"Saving game file: {fileName}.");
             var gameFile = new GameFile();
-            _appData.Game.CopyTo(gameFile);
+            _appData.Game.CopyTo(gameFile); ;
```
Comment (davilimap) [inline ArkhamOverlay/Services/GameFileService.cs]: Remove extra `;`

Comment (slyjeff) [reply ArkhamOverlay/Services/GameFileService.cs]: fixed

File: ArkhamOverlay/Services/GameFileService.cs (hunk 3 above)
Comment (slyjeff) [reply ArkhamOverlay/Services/GameFileService.cs]: fixed

File: ArkhamOverlay/Services/GameFileService.cs (hunk 2 above)
Comment (slyjeff) [reply ArkhamOverlay/Services/GameFileService.cs]: fixed

File: ArkhamOverlay/Data/Game.cs (hunk 1 above)
Comment (slyjeff) [reply ArkhamOverlay/Data/Game.cs]: fixed

**Review by davilimap — APPROVED** [review]
